services/net/transcription/TranscriptionManager.cs

```diff
@@ -0,0 +1,252 @@
+            {
+                try
+                {
+                    // TODO: Handle e-tag.
+                    // TODO: Update on some kind of interval.
+                    this.Lookups = await _api.GetLookupsAsync();
+
+                    // Listen to every enabled data source with a topic.
+                    var topics = !String.IsNullOrWhiteSpace(_options.Topics) ? _options.GetTopics() : this.Lookups?.DataSources
+                        .Where(ds => ds.IsEnabled &&
+                            ds.ContentTypeId > 0 &&
+                            !String.IsNullOrWhiteSpace(ds.Topic) &&
+                            ds.Connection.ContainsKey("import") &&
+                            ((JsonElement)ds.Connection["import"]).GetBoolean()).Select(ds => ds.Topic).ToArray() ?? Array.Empty<string>();
+
+                    if (topics.Length != 0)
+                    {
+                        var tps = String.Join(',', topics);
+                        this.Logger.LogInformation("Consuming topics: {tps}", tps);
+
+                        // TODO: Need to learn how to safely stop listening without losing content.  Every time a service goes down it will most likely lose content.
+                        await this.Consumer.ListenAsync<string, SourceContent>(HandleMessageAsync, topics);
+
+                        // Successful run clears any errors.
+                        this.State.ResetFailures();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    this.Logger.LogError(ex, "Service had an unexpected failure.");
+                    this.State.RecordFailure();
+                }
+            }
+
+            // The delay ensures we don't have a run away thread.
+            this.Logger.LogDebug("Service sleeping for {delay} ms", delay);
+            await Task.Delay(delay);
+        }
+    }
+
+    /// <summary>
+    /// Import the content.
+    /// Copy any file associated with source content.
+    /// </summary>
+    /// <param name="result"></param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    private async Task HandleMessageAsync(ConsumeResult<string, SourceContent> result)
+    {
+        if (this.State.Status != ServiceStatus.Running)
+        {
+            this.Consumer.Stop();
+            this.State.Stop();
+        }
+
+        this.Logger.LogInformation("Importing Content from Topic: {Topic}, Uid: {Key}", result.Topic, result.Message.Key);
+
+        // TODO: Failures after receiving the message from Kafka will result in missing content.  Need to handle this scenario.
+        // TODO: Handle e-tag.
+        var source = await _api.GetDataSourceAsync(result.Value.Source) ?? throw new InvalidOperationException($"Failed to fetch data source for '{result.Topic}'");
+
+        if (source.ContentTypeId == null) throw new InvalidOperationException($"Data source not configured to import content correctly");
```

Comment:
I think we should let the Content Service create content and not mix functionality here.  This service should only update content with the transcription. I think the Content Service will need to be updated so that publishes a message to Kafka.  Then the Transcription Service can listen to that specific Topic.

Reply:
Fixed.